terraform/modules/s3-bucket/main.tf

```diff
@@ -0,0 +1,15 @@
+resource "aws_s3_bucket" "s3_bucket" {
+  bucket = "${var.bucket_name}"
+  acl    = "private"
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [{
+    "Effect": "Allow",
+    "Principal": {"AWS": "arn:aws:iam::${var.access_account_id}:root"},
+    "Action": "*",
```

Comment:
The console uses the same IAM permissions.

The list permission changes the behaviour of non-existing items in the bucket:
 - if you don't have listing permission then a non-existing item responds with 403, if you have it it's the expected 404.